src/Microsoft.AspNet.Routing/Template/TemplateBinder.cs

```diff
@@ -472,6 +479,12 @@ public bool Accept(string value)
                 return true;
             }
 
+            public void Remove(string literal)
+            {
+                int startIndex = _uri.Length - literal.Length;
+                _uri.Remove(startIndex, literal.Length);
+            }
```

Comment:
`_uri.Length -= literal.Length` will do what you want. This just moves back the 'end' pointer of the buffer.